src/quo2/components/record_audio/record_audio/buttons/record_button_big.cljs

```diff
@@ -45,129 +45,131 @@
+                             ;; GitHub issue [#14561]:
+                             ;; https://github.com/status-im/status-mobile/issues/14561
+                             (reset! clear-timeout
+                               (js/setTimeout
+                                (fn []
+                                  (set-value scale scale-to-each)
+                                  (animate-linear-with-delay-loop scale
+                                                                  scale-to-total
+                                                                  signal-anim-duration-2
+                                                                  0))
+                                signal-anim-duration)))
+           stop-animation (fn []
+                            (set-value opacity 0)
+                            (reanimated/cancel-animation scale)
+                            (set-value scale 1)
+                            (when @clear-timeout (js/clearTimeout @clear-timeout)))
+           start-y-animation (fn []
+                               (reset! record-button-at-initial-position? false)
+                               (reset! record-button-is-animating? true)
+                               (animate-easing translate-y -64 250)
+                               (animate-linear-with-delay icon-opacity 0 33.33 76.66)
+                               (js/setTimeout (fn []
+                                                (reset! record-button-is-animating? false)
+                                                (when-not @touch-active? (complete-animation)))
+                                              250))
+           reset-y-animation (fn []
+                               (animate-easing translate-y 0 300)
+                               (animate-linear icon-opacity 1 500)
+                               (js/setTimeout (fn []
+                                                (reset! record-button-at-initial-position? true))
+                                              500))
+           start-x-animation (fn []
+                               (reset! record-button-at-initial-position? false)
+                               (reset! record-button-is-animating? true)
+                               (animate-easing translate-x -64 250)
+                               (animate-linear-with-delay icon-opacity 0 33.33 76.66)
+                               (animate-linear red-overlay-opacity 1 33.33)
+                               (js/setTimeout (fn []
+                                                (reset! record-button-is-animating? false)
+                                                (when-not @touch-active? (complete-animation)))
+                                              250))
+           reset-x-animation (fn []
+                               (animate-easing translate-x 0 300)
+                               (animate-linear icon-opacity 1 500)
+                               (animate-linear red-overlay-opacity 0 100)
```

Comment:
@ilmotta @Parveshdhull another dev who felt the need to create their own abstraction of reanimated methods. Without a better abstraction https://github.com/status-im/status-mobile/issues/15176 we would end up with multiple reanimated helper/utils files in the project.

Reply:
Yep, I think we should go back to the drawing board and see the kinds of problems a new/improved abstraction can bring.

That issue #15176 only mentions the "long name" problem @OmarBasem. I think we should take the opportunity to describe more problems in that issue. Only once the problems are clear we should attempt to evolve the current animation abstractions.

I have limited input for hand-written animations in status-mobile because I mostly used simpler transitions at this point, but I can try to help with some more general *code design* stuff. But right now I don't understand exactly the kinds of problems you guys are facing, so it'd be good to have more insights in issue #15176.

Reply:
@briansztamfater I will be working on an issue soon to improve the abstraction of reanimated. If you have any suggestions feel free to share it in that issue: https://github.com/status-im/status-mobile/issues/15176